### PR TITLE
Bring back TextInput.State, deprecate focusTextInput and blurTextInput

### PR DESCRIPTION
**Pull request by janicduplessis**
a275eac56e21b8f3631bcd03ace86c07a3b0c0ba removed TextInput.State but we should keep it as it was a public-ish API and we don't have any migration plan off it. Also bring back `focusTextInput` and `blurTextInput` with a deprecation warning.

## Test Plan

Tested TextInput.State is back

## Release Notes

[GENERAL][ENHANCEMENT][TextInput] - Bring back TextInput.State, deprecate focusTextInput and blurTextInput

### CODE REVIEW

**Comment by janicduplessis** [comment]
@sophiebits Could you have a look a this?

File: Libraries/Components/TextInput/TextInput.js [hunk 1]
```diff
@@ -192,7 +192,13 @@ const DataDetectorTypes = [
 
 const TextInput = createReactClass({
   displayName: 'TextInput',
-
+  statics: {
+    State: {
+      currentlyFocusedField: TextInputState.currentlyFocusedField,
+      focusTextInput: TextInputState.focusTextInput,
```
[the remark this answers is not in the document]
Comment (janicduplessis) [reply Libraries/Components/TextInput/TextInput.js]: It is usually used in combination with currentlyFocusedField, which is the node handle. There is also `findNodeHandle`. I agree the other API is preferable, but there might be valid use cases for TextInput.State one, like the keyboard library I noticed the issue in. If we still want to remove this we should probably deprecate it for a release cycle first.

[the remark this answers is not in the document]
Comment (sophiebits) [reply Libraries/Components/TextInput/TextInput.js]: We have Keyboard.dismiss (or dismissKeyboard) for blurring the focused field.

If you will take care of deprecating it then removing it in the next release, we can add it back.

[the remark this answers is not in the document]
Comment (janicduplessis) [reply Libraries/Components/TextInput/TextInput.js]: Makes sense, The only one that there is no alternative currently is `currentlyFocusedField`, which is useful to measure where the focused the input is in the screen (example: https://github.com/APSL/react-native-keyboard-aware-scroll-view/blob/master/lib/KeyboardAwareHOC.js#L267). I will add a deprecation warning to the other 2 methods.

**Comment by janicduplessis** [comment]
Split the PR, this is now only the TextInput.State part, see https://github.com/facebook/react-native/pull/19255 for the other.

**Comment by sophiebits** [comment]
:+1: thanks!

**Pull request closed without merging**